src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs

```diff
@@ -199,6 +199,11 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
             var result = base.RemoveReturns();
             foreach (var pending in result)
             {
+                if (pending.Branch is null)
```

Comment:
When does this happen?

Reply:
In a local function that is also an iterator, there is a silent generated branch out of the method before the first statement to return the enumerator.